pkg/bootstrap/bootstrap_config.go

```diff
@@ -379,6 +383,12 @@ func writeBootstrapForPlatform(config *meshconfig.ProxyConfig, node string, epoc
 	// Support passing extra info from node environment as metadata
 	meta := getNodeMetaData(localEnv, platEnv)
 
+	// Make sure the metadata is also available as part of the bootstrap generation, which was the original
+	// behavior. This avoids the need to add redundant code.
+	for k, v := range meta {
```

Comment:
This should be namespaced. perhaps `opts["meta_"+k] = v`
This is blindly overwriting other specific keys like zone, region, cluster etc.

Reply:
I was trying to restore the original behavior, but you're probably right. 
I'm not entirely sure if it's a bug or feature - I am checking if the entry doesn't exist now.

Reply:
maybe just make this a map, then in the bootstrap they can do things like `{{meta."istio.io/metadata".labels}}`? I agree it does seem like we could accidentally override some fields